File: docs/snippets/all/archetypes/line_strips3d_segments_simple.cpp
// Log a simple set of line segments.

#include <dalaran.hpp>

#include <array>
#include <vector>

int main(int argc, char* argv[]) {
    const auto rec = dalaran::RecordingStream("dalaran_example_line_segments3d");
    rec.spawn().exit_on_failure();

    std::vector<std::vector<std::array<float, 3>>> points = {
        {{0.f, 0.f, 0.f}, {0.f, 0.f, 1.f}},
        {{1.f, 0.f, 0.f}, {1.f, 0.f, 1.f}},
        {{1.f, 1.f, 0.f}, {1.f, 1.f, 1.f}},
        {{0.f, 1.f, 0.f}, {0.f, 1.f, 1.f}},
    };

    rec.log("segments", dalaran::LineStrips3D(points));
}

File: docs/snippets/all/archetypes/line_strips3d_segments_simple.rs
//! Log a simple set of line segments.

fn main() -> Result<(), Box<dyn std::error::Error>> {
    let rec =
        dalaran::RecordingStreamBuilder::new("dalaran_example_line_segments3d")
            .spawn()?;

    let points = [
        [0., 0., 0.],
        [0., 0., 1.],
        [1., 0., 0.],
        [1., 0., 1.],
        [1., 1., 0.],
        [1., 1., 1.],
        [0., 1., 0.],
        [0., 1., 1.],
    ];
    rec.log("segments", &dalaran::LineStrips3D::new(points.chunks(2)))?;

    Ok(())
}
